documentation/content/content-nodes.html

```diff
@@ -213,94 +217,69 @@ <h3 id="master-election">Master election</h3>
-Each incoming document is assigned to a bucket and forwarded to the right search nodes.
-All requests related to these buckets are sent through the distributors.
+Document operations are mapped to content nodes based on bucket locations -
+each put/update/get/remove is mapped to a <a href="buckets.html">bucket</a>
+and sent to the right content nodes.
+To manage the document set as it grows and nodes change, buckets move.
+I.e, sets of documents (a bucket) is copied between nodes for redundancy /
+redistribution for growth / node failures.
 </p><p>
-A <a href="../elastic-vespa.html#bucket-distribution-algorithm">distribution algorithm</a>
-calculates what distributors are responsible for given buckets based on
-information of what nodes are available in the cluster.
-Clients calculate what distributor to talk to using information in the cluster state.
+Buckets are mapped to distributors using the <a href="idealstate.html">ideal state algorithm</a>.
+As the cluster state changes, buckets are re-mapped immediately.
+The mapping does not overlap - a bucket is owned by one distributor.
 </p><p>
-When a distributor goes down or becomes available,
-all distributors reshuffle what buckets they are responsible for.
-During this reshuffling, clients may get some requests temporarily failing
-as distributors taking over bucket ownership doesn't yet know where copies of its buckets are located.
-These requests will automatically be resent, providing timeouts allow for it.
-Changes in what distributors are up and available will currently cause windows of availability loss.
-Other distributors take over bucket ownership.
-To do this, they fetch bucket metadata for new buckets from all storage nodes,
-in-memory operations for speed.
+Distributors do not persist the bucket database,
+the bucket-to-content-node mapping is stored in memory in the distributor.
+Document count per bucket is stored as well. <!-- ToDo: checksum? -->
+At distributor (re)start, content nodes are polled for bucket information,
+and return which buckets are owned by this distributor (using the ideal state algorithm).
+There is hence no master / name node in Vespa.
+Likewise, at any distributor cluster state change,
+content nodes are polled for bucket handover -
+a distributor will then handle a new set of buckets.
 </p><p>
-The distribution algorithm also calculates what content nodes copies of buckets should be stored on.
-As moving bucket copies take time, the distributors track current state in a bucket database,
-allowing requests to be processed independent of where buckets are currently located.
-Distributors use the distribution algorithm to detect buckets that are stored on wrong nodes,
-and move copies to correct nodes when there are free resources to do so.
+Document API clients do not interface with the cluster controller,
+and does hence not know the cluster start at startup.
+A random distributor is used first. <!-- ToDo: Or does it use the ideal state algo? -->
```

Comment:
When a client starts up it does not have the cluster state information required to use the ideal state algorithm, so it has no choice other than to send to random node in the config (which it does have).